cfme/infrastructure/provider/__init__.py

```diff
@@ -286,10 +292,74 @@ def hosts(self):
         return result
 
 
+@navigator.register(Provider, 'All')
+class All(CFMENavigateStep):
+    prerequisite = NavigateToAttribute('appliance', 'LoggedIn')
+
+    def step(self):
+        from cfme.web_ui.menu import nav
+        nav._nav_to_fn('Compute', 'Infrastructure', 'Providers')(None)
+
+
+@navigator.register(Provider, 'Add')
+class Add(CFMENavigateStep):
+    prerequisite = NavigateToSibling('All')
+
+    def step(self):
+        cfg_btn('Add a New Infrastructure Provider')
+
+
+@navigator.register(Provider, 'Discover')
+class Discover(CFMENavigateStep):
+    prerequisite = NavigateToSibling('All')
+
+    def step(self):
+        cfg_btn('Discover Infrastructure Providers')
+
+
+@navigator.register(Provider, 'Details')
+class Details(CFMENavigateStep):
+    def prerequisite(self):
+        NavigateToSibling('All')
+        # Make sure no quadicons are selected and default to grid view
+        # Can we assume these two things at this level? (comment to be removed before merge)
+        # If this should be actively set, a utility function will be needed to shorten this
+        tb.select("Grid View")
+        sel.check(paginator.check_all())
+        sel.uncheck(paginator.check_all())
+
+
+    def step(self):
+        sel.click(Quadicon(self.obj.name, 'infra_prov'))
+
+
+@navigator.register(Provider, 'ManagePolicies')
+class ManagePolicies(CFMENavigateStep):
+    def prerequisite(self):
```

Comment:
Understood - I had skel'd them out as functions because of the resetting, will switch to 1-line usage now since at the very least it will already be in the 'All' destination.